authors/apps/authentication/views.py

```diff
@@ -127,6 +127,22 @@ def retrieve(self, request, *args, **kwargs):
     def update(self, request, *args, **kwargs):
         serializer_data = request.data.get('user', {})
 
+        serializer_data = {
+            'username': serializer_data.get('username', request.user.username),
+            'email': serializer_data.get('email', request.user.email),
```

Comment:
Why do you set them one by one instead of passing an instance and request data to the serializer?For example, by having `serializer = serializer_class(user, serializer_data)?`

Reply:
@meshack-mbuvi I did that because I need to capture the `profile` key set in the UserSerializer. `i.e profile_data = validated_data.pop('profile')`